kubeflow/katib/modeldb.libsonnet

```diff
@@ -0,0 +1,252 @@
+      metadata: {
+        name: "modeldb-backend",
+        namespace: namespace,
+        labels: {
+          app: "modeldb",
+          component: "backend",
+        },
+      },
+      spec: {
+        type: "ClusterIP",
+        ports: [
+          {
+            port: 6543,
+            protocol: "TCP",
+            name: "api",
+          },
+        ],
+        selector: {
+          app: "modeldb",
+          component: "backend",
+        },
+      },
+    },  // backendService
+    backendDeployment: {
+      apiVersion: "extensions/v1beta1",
+      kind: "Deployment",
+      metadata: {
+        labels: {
+          app: "modeldb",
+          component: "backend",
+        },
+        name: "modeldb-backend",
+        namespace: namespace,
+      },
+      spec: {
+        replicas: 1,
+        template: {
+          metadata: {
+            labels: {
+              app: "modeldb",
+              component: "backend",
+            },
+            name: "modeldb-backend",
+          },
+          spec: {
+            containers: [
+              {
+                args: [
+                  "modeldb-db",
+                ],
+                image: "mitdbg/modeldb-backend:latest",
+                name: "modeldb-backend",
+                ports: [
+                  {
+                    containerPort: 6543,
+                    name: "api",
+                  },
+                ],
+              },
+            ],
+          },
+        },
+      },
+    },  // backendDeployment
+
+    dbService: {
+      apiVersion: "v1",
+      kind: "Service",
+      metadata: {
+        labels: {
+          app: "modeldb",
+          component: "db",
+        },
+        name: "modeldb-db",
+        namespace: namespace,
+      },
+      spec: {
+        ports: [
+          {
+            name: "dbapi",
+            port: 27017,
+            protocol: "TCP",
+          },
+        ],
+        selector: {
+          app: "modeldb",
+          component: "db",
+        },
+        type: "ClusterIP",
+      },
+    },  // dbService
+
+    dbDeployment: {
+      apiVersion: "extensions/v1beta1",
+      kind: "Deployment",
+      metadata: {
+        labels: {
+          app: "modeldb",
+          component: "db",
+        },
+        name: "modeldb-db",
+        namespace: namespace,
+      },
+      spec: {
+        replicas: 1,
+        template: {
+          metadata: {
+            labels: {
+              app: "modeldb",
+              component: "db",
+            },
+            name: "modeldb-db",
+          },
+          spec: {
+            containers: [
+              {
+                image: "mongo:3.4",
+                name: "modeldb-db",
+                ports: [
+                  {
+                    containerPort: 27017,
+                    name: "dbapi",
+                  },
+                ],
+              },
+            ],
+          },
+        },
+      },
+    },  // dbDeployment
+
+    frontendService: {
```

Comment:
Either in this PR or a follow on one we should add an Ambassador prefix to access the front end.

Reply:
I plan to do that in next PR. We have a different issue https://github.com/kubeflow/katib/issues/88